net/src/Sails.Remoting/EventListenerExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿using EnsureThat;
+using Sails.Remoting.Abstractions.Core;
+using Substrate.Gear.Api.Generated.Model.gprimitives;
+using Substrate.NetApi.Model.Types;
+
+namespace Sails.Remoting;
+
+public static class EventListenerExtensions
+{
+    /// <summary>
+    /// Projects Gear event to Typed Service Event
+    /// </summary>
+    public static EventListener<(ActorId Source, T Event)> SelectEvents<T>(
```

Comment:
I would go w/o this extension:
- its name doesn't sound right to me as it is not a selector but a constructor of a new type, i.e. it is more like `ToServiceEventListener`
- this is a one liner which is used in a single place